### PR TITLE
feat(vd):  support filesystem mode

### PR DESCRIPTION
**Pull request by yaroslavborbat**
## Description
Support filesystem mode.
Support qcow2 images.

## Checklist
- [ ] The code is covered by unit tests.
- [X] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: images/virtualization-artifact/pkg/controller/kvbuilder/dv.go [hunk 1]
```diff
@@ -56,24 +63,38 @@ func NewDV(name types.NamespacedName) *DV {
...
+		format = unpackFormatRaw
+	case corev1.PersistentVolumeFilesystem:
+		format = unpackFormatQcow2
+	}
+	annos := b.Resource.ObjectMeta.GetAnnotations()
+	if annos == nil {
+		annos = map[string]string{}
+	}
+	annos[cc.AnnUnpackFormat] = format
+	b.Resource.ObjectMeta.SetAnnotations(annos)
```
Comment (diafour) [inline images/virtualization-artifact/pkg/controller/kvbuilder/dv.go:95]:
```suggestion
	cc.AddAnnotation(b.Resource, cc.AnnUnpackFormat, format)
```

File: images/virtualization-artifact/pkg/common/pvc/util.go [hunk 2]
```diff
@@ -26,40 +26,32 @@ import (
...
+	size resource.Quantity,
+	accessMode corev1.PersistentVolumeAccessMode,
+	volumeMode corev1.PersistentVolumeMode,
+) *corev1.PersistentVolumeClaimSpec {
 	if os.Getenv(ForceDVReadWriteOnceVar) == "yes" {
-		return CreateSpecReadWriteOnce(storageClassName, size)
+		return CreateSpec(storageClassName, size, corev1.ReadWriteOnce, volumeMode)
 	}
-	return CreateSpecReadWriteMany(storageClassName, size)
+	return CreateSpec(storageClassName, size, accessMode, volumeMode)
```
Comment (diafour) [inline images/virtualization-artifact/pkg/common/pvc/util.go:56]: FORCE_DV_READ_WRITE_ONCE is not needed anymore as accessClass is now calculated in parseVolumeMode and ReadWriteOnce is a default accessMode. You can remove this const from this file and also from virtualization-controller envs template helper.

File: images/cdi-artifact/patches/README.md [hunk 3]
```diff
@@ -34,3 +34,11 @@ Do not manage DataVolume CRD with cdi-operator. Module will install this CRD usi
...
 
 Set the storage class name for the scratch pvc from the original pvc that will own the scratch pvc, or set it to an empty value if not available.
+
+#### `012-add-caps-for-deckhouse-provisioners.patch`
+
+Add capabilities for deckhouse provisioners to cdi StorageProfile.
+
+#### `013-converting-images-in-filesystem-to-qcow2.patch`
+
+Converting images in the file system to qcow2.
```
Comment (diafour) [inline images/cdi-artifact/patches/README.md:44]: This patch deserves more details about what is added and how it works.

File: images/virt-artifact/patches/021-support-qcow2-for-filesystem.patch [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+ 	}
+@@ -733,7 +732,7 @@ func Convert_v1_Hotplug_DataVolume_To_api_Disk(name string, disk *api.Disk, c *C
+ // Convert_v1_FilesystemVolumeSource_To_api_Disk takes a FS source and builds the domain Disk representation
+ func Convert_v1_FilesystemVolumeSource_To_api_Disk(volumeName string, disk *api.Disk, volumesDiscardIgnore []string) error {
+ 	disk.Type = "file"
+-	disk.Driver.Type = "raw"
++	disk.Driver.Type = "qcow2"
+ 	disk.Driver.ErrorPolicy = "stop"
+ 	disk.Source.File = GetFilesystemVolumePath(volumeName)
+ 	if !contains(volumesDiscardIgnore, volumeName) {
```
Comment (diafour) [inline images/virt-artifact/patches/021-support-qcow2-for-filesystem.patch:79]: So, we change raw to qcow2 and there is no way to use "sparse raw" disks on filesystem? It is a significant change in CDI behaviour and should be documented in patch description in README.md.

**Review by diafour — DISMISSED** [review]

**Review by fl64 — APPROVED** [review]

**Review by Isteb4k — APPROVED** [review]

**Pull request merged**